### DISCUSSION TITLE
Pull request/commit rules

### DISCUSSION THREAD

**Discussion by GeorgWa** [MannLabs/alphabase#50] [General]
Hi,

what nbdev functions should I run before creating pull requests?

I always run nbdev_prepare and nbdev_preview which results in the documentation being updated and the commit becoming overloaded with documentation updates.
Can I somehow clean up the created documentation, quarto.yml file etc. or should I remove it from the commit?

Best,
Georg

**Comment by jalew188** [comment]
nbdev is not the core of alphabase and alphapeptdeep any more. But we still use it for unit testing:

- run `nbdev_clean` to clean notebook ids.
- run `nbdev_test` for unit tests.
